taca/delivery/delivery_classes.py

```diff
@@ -111,3 +137,251 @@ def stage_data(self):
+                f"Project {self.project_id} has more than one entry in StatusDB orderportal_db"
+            )
+        portal_id = rows[0].value
+        # Get project info from order portal API
+        get_project_url = "{}/v1/order/{}".format(
+            self.orderportal.get("orderportal_api_url"), portal_id
+        )
+        headers = {
+            "X-OrderPortal-API-key": self.orderportal.get("orderportal_api_token")
+        }
+        response = requests.get(get_project_url, headers=headers)
+        if response.status_code != 200:
+            raise AssertionError(
+                "Status code returned when trying to get "
+                "project info from the order portal: "
+                f"{portal_id} was not 200. Response was: {response.content}"
+            )
+        return json.loads(response.content)
+
+    def get_pi_email(self, given_pi_email):
+        """Determine the PI email."""
+        if given_pi_email:
+            logger.warning(
+                f"PI email for project {self.project_id} specified by user: {given_pi_email}"
+            )
+            return given_pi_email
+        else:
+            found_pi_email = self.order_details["fields"]["project_pi_email"]
+            logger.info(
+                f"PI email for project {self.project_id} found: {found_pi_email}"
+            )
+            return found_pi_email
+
+    def get_other_member_details(
+        self, other_member_emails=[], ignore_orderportal_members=False
+    ):
+        """Set other contact details if available. This is not mandatory so
+        the method will not raise error if it could not find any contact
+        """
+        other_member_details = []
+        if not ignore_orderportal_members:
+            logger.info("Fetching additional members from order portal.")
+            try:
+                owner_email = self.order_details.get("owner", {}).get("email")
+                if (
+                    owner_email
+                    and owner_email != self.pi_email
+                    and owner_email not in other_member_details
+                ):
+                    other_member_details.append(owner_email)
+                bioinfo_email = self.order_details.get("fields", {}).get(
+                    "project_bx_email"
+                )
+                if (
+                    bioinfo_email
+                    and bioinfo_email != self.pi_email
+                    and bioinfo_email not in other_member_details
+                ):
+                    other_member_details.append(bioinfo_email)
+                lab_email = self.order_details.get("fields", {}).get(
+                    "project_lab_email"
+                )
+                if (
+                    lab_email
+                    and lab_email != self.pi_email
+                    and lab_email not in other_member_details
+                ):
+                    other_member_details.append(lab_email)
+            except (AssertionError, ValueError):
```

Comment:
Where is this exception thrown? What happens if the exception is thrown on bioinfo_email but it also has lab_email?

Reply:
Good question... I copied this one over from taca-ngi-pipeline, I agree that it looks a bit unstable. I'll give it another look 👍